crates/dprint/src/cli/run_cli.rs

```diff
@@ -266,69 +306,33 @@ fn run_editor_service<TEnvironment: Environment>(
-        let new_file_paths = if !should_reattempt || &new_file_patterns != file_patterns || &new_absolute_paths != absolute_paths {
-            resolve_file_paths(&new_file_patterns, &new_absolute_paths, &config, environment)?
-        } else {
-            file_paths.clone()
-        };
-
-        // canonicalize the file path, then check if it's in the list of file paths.
-        match environment.canonicalize(&file_path) {
-            Ok(resolved_file_path) => {
-                messenger.send_message(if new_file_paths.contains(&resolved_file_path) { 1 } else { 0 }, Vec::new())?;
-                Ok((new_file_patterns, new_absolute_paths, new_file_paths))
-            }
-            Err(err) => {
-                // try one more time by re-resolving filepaths
-                if should_reattempt {
-                    return handle_check_path(messenger, args, cache, environment, &new_file_patterns, &new_absolute_paths, &new_file_paths, false);
-                }
-                environment.log_error(&format!("Error canonicalizing file {}: {}", file_path.display(), err.to_string()));
-                messenger.send_message(0, Vec::new())?; // don't format, something went wrong
-                Err(err)
-            },
-        }
-    }
-
     loop {
         let message_kind = messenger.read_code()?;
         match message_kind {
             // shutdown
             0 => return Ok(()),
             // check path
             1 => {
-                let (new_file_patterns, new_absolute_paths, new_file_paths) = handle_check_path(
-                    &mut messenger,
-                    args,
-                    cache,
-                    environment,
-                    &file_patterns,
-                    &absolute_paths,
-                    &file_paths,
-                    true,
-                )?;
-                file_patterns = new_file_patterns;
-                absolute_paths = new_absolute_paths;
-                file_paths = new_file_paths;
+                let file_path = messenger.read_single_part_path_buf_message()?;
+                // update the glob file patterns and absolute paths by re-retrieving the config file
+                let config = resolve_config_from_args(args, cache, environment)?;
+                let (include_globset, exclude_globset) = build_include_exclude_glob_sets(&config, args, environment)?;
+
+                // canonicalize the file path, then check if it's in the list of file paths.
+                match environment.canonicalize(&file_path) {
+                    Ok(resolved_file_path) => {
+                        let matches_includes = include_globset.is_match(&resolved_file_path);
+                        let matches_excludes = exclude_globset.is_match(&resolved_file_path);
+                        let file_exists = environment.path_exists(PathBuf::new().join(config.base_path).join(file_path).as_path());
```

Comment:
TODO: currently assumes that `file_path` is relative  to the base path
(also need to add a test for this)

Reply:
I think checking `matches_includes` and `matches_excludes` with `resolved_file_path` should be sufficient. This would allow people to format a file in the editor when they haven't saved the file yet (which is not possible today)

Reply:
I added this because it started failing the existing editor service tests, on the `/nonexistent.txt` file. Can make this change if you're happy with it 👍

Reply:
That would be good! It's probably a test that shouldn't exist.